### PR TITLE
Fixes include path broken by virtual pip packaging

### LINKED ISSUE #1129: Nightly TF 1.14.1.dev20190607 breaks v0.16.3
**Environment:**
1. Framework: TensorFlow 
2. Framework version: 1.14.1.dev20190607
3. Horovod version: 0.16.3
4. MPI version: 2.0.2
5. CUDA version: 10.0
6. NCCL version: 2.4.2
7. Python version: 3.5m
8. OS and version: Debian 9.8

**Bug report:**

Build Horovod with latest TF nightly fails with the following error:

```
  x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fdebug-prefix-map=/build/python3.5-3.5.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -DEIGEN_MPL2_ONLY=1 -Ithird_party/eigen -Ithird_party/lbfgs/include -Ithird_party/boost/assert/include -Ithird_party/boost/config/include -Ithird_party/boost/core/include -Ithird_party/boost/detail/include -Ithird_party/boost/iterator/include -Ithird_party/boost/lockfree/include -Ithird_party/boost/mpl/include -Ithird_party/boost/parameter/include -Ithird_party/boost/predef/include -Ithird_party/boost/preprocessor/include -Ithird_party/boost/static_assert/include -Ithird_party/boost/type_traits/include -Ithird_party/boost/utility/include -Ithird_party/flatbuffers/include -I/usr/include/python3.5m -I/home/byronyi/.virtualenv/tf/include/python3.5m -c horovod/tensorflow/mpi_ops.cc -o build/temp.linux-x86_64-3.5/horovod/tensorflow/mpi_ops.o -std=c++11 -fPIC -O2 -Wall -I/usr/lib/x86_64-linux-gnu/openmpi/include/openmpi -I/usr/lib/x86_64-linux-gnu/openmpi/include/openmpi/opal/mca/event/libevent2022/libevent -I/usr/lib/x86_64-linux-gnu/openmpi/include/openmpi/opal/mca/event/libevent2022/libevent/include -I/usr/lib/x86_64-linux-gnu/openmpi/include -pthread -L/usr//lib -L/usr/lib/x86_64-linux-gnu/openmpi/lib -lmpi_cxx -lmpi -I/home/byronyi/.virtualenv/tf/lib/python3.5/site-packages/tensorflow/include -D_GLIBCXX_USE_CXX11_ABI=0
  cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
  horovod/tensorflow/mpi_ops.cc:22:42: fatal error: tensorflow/core/framework/op.h: No such file or directory
   #include "tensorflow/core/framework/op.h"
                                            ^
  compilation terminated.
  INFO: Unable to build TensorFlow plugin, will skip it.
```

### PR DESCRIPTION
**Pull request by byronyi**
Ping @mihaimaruseac for review. Ping @lark as this is a work-related contribution.

This fixes https://github.com/horovod/horovod/issues/1129 in our in-house nightly build.

### CODE REVIEW

File: tensorflow/tools/pip_package/build_pip_package.sh [hunk 1]
```diff
@@ -176,6 +176,18 @@ function build_wheel() {
...
   echo $(date) : "=== Building wheel"
   "${PYTHON_BIN_PATH:-python}" setup.py bdist_wheel ${PKG_NAME_FLAG} >/dev/null
+
+  # Fix include path by moving tensorflow_core/include/tensorflow{_core,}
+  WHEEL_TMPDIR=$(mktemp -d)
+  WHEEL=$(readlink -f dist/*.whl)
+  unzip $WHEEL -d $WHEEL_TMPDIR > /dev/null
+  rm $WHEEL
+  pushd $WHEEL_TMPDIR > /dev/null
+  mv tensorflow_core/include/tensorflow{_core,}
```
Comment (mihaimaruseac) [inline tensorflow/tools/pip_package/build_pip_package.sh]: I think we only need this line but immediately after line 172 above:

```bash
mv tensorflow tensorflow_core   # this is line 172
mv tensorflow_core/include/tensorflow{_core,}   # this is the line
```

This way we don't unzip and rezip the wheel and potentially breaking the contract in manifest.

Comment (mihaimaruseac) [reply tensorflow/tools/pip_package/build_pip_package.sh]: I'm going to approve as is for now and fix that tomorrow after we have new pips

**Review by mihaimaruseac — APPROVED** [review]

File: tensorflow/tools/pip_package/build_pip_package.sh (hunk 1 above)
Comment (byronyi) [reply tensorflow/tools/pip_package/build_pip_package.sh]: That’s weird, why do we have them under tensorflow_core in the first place, before bdist_wheel? I’ll take a closer look at your suggested fix.

Comment (byronyi) [reply tensorflow/tools/pip_package/build_pip_package.sh]: > This way we don't unzip and rezip the wheel and potentially breaking the contract in manifest.

The MANIFEST.in file seems long broken anyway. Do we need to fix them as well?

**Comment by byronyi** [comment]
Oh this should not be merged...ping @perfinion to help sort out this patch in a proper way.

**Pull request merged**